node/core/dispute-coordinator/src/lib.rs

```diff
@@ -349,6 +351,103 @@ where
 	}
 }
 
+// Restores the subsystem's state before proceeding with the main event loop. Primarily, this
+// repopulates the rolling session window the relevant session information to handle incoming
+// import statement requests.
+//
+// This method also retransmits a DisputeParticiationMessage::Participate for any non-concluded
+// disputes for which the subsystem doesn't have a local statement, ensuring it eventually makes an
+// arbitration on the disupte.
+async fn resume<B, Context>(
+	ctx: &mut Context,
+	state: &mut State,
+	backend: &mut B,
+) -> Result<(), Error>
+where
+	Context: overseer::SubsystemContext<Message = DisputeCoordinatorMessage>,
+	Context: SubsystemContext<Message = DisputeCoordinatorMessage>,
+	B: Backend,
+{
+
+	// Query for the set of active leaves to reconstruct our rolling session window.
+	let (tx, rx) = oneshot::channel();
+	ctx.send_message(ChainSelectionMessage::Leaves(tx)).await;
```

Comment:
Is this the proper way to reconstruct a `RollingSessionWindow`?

That is, should we fetch `Leaves` from the ChainSelection subsystem and subsequently call `handle_new_activations` for all leaves?

Reply:
I think we should instead run the `resume` logic on the first time we receive an `ActivatedLeaf` from the overseer. The overseer gives all current leaves on startup. We shouldn't need any dependency on `ChainSelection`